plugins/node/opentelemetry-instrumentation-aws-sdk/test/aws-sdk-v2.test.ts

```diff
@@ -275,9 +277,21 @@ describe('instrumentation-aws-sdk-v2', () => {
         const awsSpans = getAwsSpans();
         expect(awsSpans.length).toBe(1);
         const [spanCreateBucket] = awsSpans;
-        expect(spanCreateBucket.attributes[AttributeNames.AWS_ERROR]).toBe(
-          responseMockWithError.error
+        const exceptionEvent = spanCreateBucket.events.filter(
+          event => event.name === ExceptionEventName
         );
+        //need to handle test differently if more than one exception
+        expect(exceptionEvent.length).toBe(1);
+
+        expect(exceptionEvent[0]).toStrictEqual(
+          expect.objectContaining({
+            name: 'exception',
+            attributes: {
+              'exception.message': 'something went wrong',
```

Comment:
This PR is about the error being an object with message and code, and not a string (which was the original motivation for the change). This test uses `responseMockWithError` which is

```js
  const responseMockWithError = {
    requestId: '0000000000000',
    error: 'something went wrong',
    httpResponse: {
      statusCode: 400,
    },
  };
```

Should we add a test where the error is something like: 
```js
{
  code: 1234,
  message: 'test error as object'
}
```

and also assert the error code in the test?